Samples/WebSignIn/WebSignIn (iOS)/WebSignIn/SignInViewController.swift

```diff
@@ -70,4 +74,69 @@ class SignInViewController: UIViewController {
             }
         }
     }
+    
+    @IBAction func signInWithRefreshToken(_ sender: Any) {
+        let pasteboard = UIPasteboard.general
+        if pasteboard.hasStrings,
+           let pasteStrings = pasteboard.strings
+        {
+            for string in pasteStrings {
+                var matchString = string.unicodeScalars
+                matchString.removeAll(where: CharacterSet.alphanumerics.contains)
+                if matchString.isEmpty {
+                    signInWith(refreshToken: string)
+                    return
+                }
+            }
+        }
+        
+        let alert = UIAlertController(title: "Sign In with Refresh Token",
+                                      message: "Enter the refresh token below",
+                                      preferredStyle: .alert)
+        alert.addTextField() { field in
+            field.placeholder = "refresh_token"
+            field.autocorrectionType = .no
+        }
+        alert.addAction(.init(title: "Cancel", style: .cancel))
+        alert.addAction(.init(title: "OK", style: .default) { action in
+            guard let textField = alert.textFields?.first,
+                let string = textField.text,
+                !string.isEmpty
+            else {
+                alert.dismiss(animated: true) {
+                    let alert = UIAlertController(title: "Invalid refresh token",
+                                                  message: nil,
+                                                  preferredStyle: .alert)
+                    alert.addAction(.init(title: "OK", style: .default))
+                    self.present(alert, animated: true)
+                }
+                return
+            }
+            
+            self.signInWith(refreshToken: string)
+        })
+        present(alert, animated: true)
+    }
+    
+    func signInWith(refreshToken: String) {
```

Comment:
Is signInWith visible to the users? Just wondering if users might be confused between signIn and signInWith, but I'm not too familiar with swift syntax here.

Reply:
This is just a part of the test app, to streamline the code.

The sample app will auto-detect if a string is copied to the clipboard, and will attempt to trigger a refresh with that.  It makes the sample flow so much smoother to demonstrate. 

Otherwise, it'll prompt the user for the refresh token in an alert dialog.